### PR TITLE
UTs: Rename ResourceTests to TestResources

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
The directory doesn't contain tests, it contains resources.

The new name is aligned with the `TestCases` directory the way it's used.

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.cs [hunk 1]
```diff
@@ -163,7 +163,7 @@ public void ProjectConfiguration_WhenFileIsMissing_ThrowException()
     [TestMethod]
     public void ProjectConfiguration_WhenInvalidXml_ThrowException()
     {
-        var sut = CreateSut(AnalysisScaffolding.CreateOptions($@"ResourceTests\SonarProjectConfig\Invalid_Xml\SonarProjectConfig.xml"));
+        var sut = CreateSut(AnalysisScaffolding.CreateOptions($@"TestResources\SonarProjectConfig\Invalid_Xml\SonarProjectConfig.xml"));
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.cs:166]:
```suggestion
        var sut = CreateSut(AnalysisScaffolding.CreateOptions(@"TestResources\SonarProjectConfig\Invalid_Xml\SonarProjectConfig.xml"));
```

File: analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.cs [hunk 2]
```diff
@@ -178,7 +178,7 @@ public void SonarLintFile_LoadsExpectedValues(string language)
     {
         var analyzerLanguage = language == "cs" ? AnalyzerLanguage.CSharp : AnalyzerLanguage.VisualBasic;
         var (compilation, _) = CreateDummyCompilation(analyzerLanguage, "ExtraEmptyFile");
-        var options = AnalysisScaffolding.CreateOptions($"ResourceTests\\SonarLintXml\\All_properties_{language}\\SonarLint.xml");
+        var options = AnalysisScaffolding.CreateOptions($@"TestResources\SonarLintXml\All_properties_{language}\SonarLint.xml");
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.cs:181]:
```suggestion
        var options = AnalysisScaffolding.CreateOptions(@"TestResources\SonarLintXml\All_properties_{language}\SonarLint.xml");
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs [hunk 3]
```diff
@@ -37,7 +37,7 @@ public class ParameterLoaderTest
         public void SetParameterValues_WithInvalidSonarLintPath_DoesNotPopulateParameters(string filePath)
         {
             // Arrange
-            var compilation = CreateCompilationWithOption(filePath, SourceText.From(File.ReadAllText("ResourceTests\\SonarLintXml\\All_properties_cs\\SonarLint.xml")));
+            var compilation = CreateCompilationWithOption(filePath, SourceText.From(File.ReadAllText("TestResources\\SonarLintXml\\All_properties_cs\\SonarLint.xml")));
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs:40]:
```suggestion
            var compilation = CreateCompilationWithOption(filePath, SourceText.From(File.ReadAllText(@"TestResources\SonarLintXml\All_properties_cs\SonarLint.xml")));
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs [hunk 4]
```diff
@@ -53,7 +53,7 @@ public void SetParameterValues_WithInvalidSonarLintPath_DoesNotPopulateParameter
         public void SetParameterValues_WithValidSonarLintPath_PopulatesProperties(string filePath)
         {
             // Arrange
-            var compilation = CreateCompilationWithOption(filePath, SourceText.From(File.ReadAllText("ResourceTests\\SonarLintXml\\All_properties_cs\\SonarLint.xml")));
+            var compilation = CreateCompilationWithOption(filePath, SourceText.From(File.ReadAllText("TestResources\\SonarLintXml\\All_properties_cs\\SonarLint.xml")));
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs:56]:
```suggestion
            var compilation = CreateCompilationWithOption(filePath, SourceText.From(File.ReadAllText(@"TestResources\SonarLintXml\All_properties_cs\SonarLint.xml")));
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs [hunk 5]
```diff
@@ -67,7 +67,7 @@ public void SetParameterValues_WithValidSonarLintPath_PopulatesProperties(string
         public void SetParameterValues_SonarLintFileWithIntParameterType_PopulatesProperties()
         {
             // Arrange
-            var compilation = CreateCompilationWithOption("ResourceTests\\SonarLintXml\\All_properties_cs\\SonarLint.xml");
+            var compilation = CreateCompilationWithOption("TestResources\\SonarLintXml\\All_properties_cs\\SonarLint.xml");
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs:70]:
```suggestion
            var compilation = CreateCompilationWithOption(@"TestResources\SonarLintXml\All_properties_cs\SonarLint.xml");
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs [hunk 6]
```diff
@@ -113,7 +113,7 @@ public void SetParameterValues_SonarLintFileWithBooleanParameterType_PopulatesPr
         public void SetParameterValues_SonarLintFileWithoutRuleParameters_DoesNotPopulateProperties()
         {
             // Arrange
-            var compilation = CreateCompilationWithOption("ResourceTests\\SonarLintXml\\All_properties_cs\\SonarLint.xml");
+            var compilation = CreateCompilationWithOption("TestResources\\SonarLintXml\\All_properties_cs\\SonarLint.xml");
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs:116]:
```suggestion
            var compilation = CreateCompilationWithOption(@"TestResources\SonarLintXml\All_properties_cs\SonarLint.xml");
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs [hunk 7]
```diff
@@ -31,7 +31,7 @@ public class ProjectConfigReaderTest
         [TestMethod]
         public void AllPropertiesAreSet()
         {
-            var sut = CreateProjectConfigReader($"ResourceTests\\SonarProjectConfig\\Path_Windows\\SonarProjectConfig.xml");
+            var sut = CreateProjectConfigReader($"TestResources\\SonarProjectConfig\\Path_Windows\\SonarProjectConfig.xml");
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs:34]:
```suggestion
            var sut = CreateProjectConfigReader(@"SonarProjectConfig\Path_Windows\SonarProjectConfig.xml");
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs [hunk 8]
```diff
@@ -44,7 +44,7 @@ public void AllPropertiesAreSet()
         [TestMethod]
         public void WhenAllValuesAreSet_LoadsExpectedValues()
         {
-            var sut = CreateProjectConfigReader($"ResourceTests\\SonarProjectConfig\\Path_Windows\\SonarProjectConfig.xml");
+            var sut = CreateProjectConfigReader($"TestResources\\SonarProjectConfig\\Path_Windows\\SonarProjectConfig.xml");
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs:47]:
```suggestion
            var sut = CreateProjectConfigReader(@"TestResources\SonarProjectConfig\Path_Windows\SonarProjectConfig.xml");
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs [hunk 9]
```diff
@@ -59,12 +59,12 @@ public void WhenAllValuesAreSet_LoadsExpectedValues()
         [DataRow("Path_Unix", @"/home/user/.sonarqube/conf/0/FilesToAnalyze.txt")]
         [DataRow("Path_Windows", @"c:\foo\bar\.sonarqube\conf\0\FilesToAnalyze.txt")]
         public void WithVariousPathFormats_ReturnsValueAsIs(string project, string expectedFilesToAnalyzePath) =>
-            CreateProjectConfigReader($"ResourceTests\\SonarProjectConfig\\{project}\\SonarProjectConfig.xml").FilesToAnalyzePath.Should().Be(expectedFilesToAnalyzePath);
+            CreateProjectConfigReader($"TestResources\\SonarProjectConfig\\{project}\\SonarProjectConfig.xml").FilesToAnalyzePath.Should().Be(expectedFilesToAnalyzePath);
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs:62]:
```suggestion
            CreateProjectConfigReader($@"TestResources\SonarProjectConfig\{project}\SonarProjectConfig.xml").FilesToAnalyzePath.Should().Be(expectedFilesToAnalyzePath);
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs [hunk 10]
```diff
@@ -59,12 +59,12 @@ public void WhenAllValuesAreSet_LoadsExpectedValues()
...
 
         [TestMethod]
         public void WhenHasMissingFilesToAnalyzePath_ReturnsCorrectValue()
         {
-            var sut = CreateProjectConfigReader($"ResourceTests\\SonarProjectConfig\\MissingFilesToAnalyzePath\\SonarProjectConfig.xml");
+            var sut = CreateProjectConfigReader($"TestResources\\SonarProjectConfig\\MissingFilesToAnalyzePath\\SonarProjectConfig.xml");
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs:67]:
```suggestion
            var sut = CreateProjectConfigReader(@"TestResources\SonarProjectConfig\MissingFilesToAnalyzePath\SonarProjectConfig.xml");
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs [hunk 11]
```diff
@@ -75,7 +75,7 @@ public void WhenHasMissingFilesToAnalyzePath_ReturnsCorrectValue()
 
         [TestMethod]
         public void WhenHasUnexpectedProjectType_FallsBackToProduct() =>
-            CreateProjectConfigReader($"ResourceTests\\SonarProjectConfig\\UnexpectedProjectTypeValue\\SonarProjectConfig.xml").ProjectType.Should().Be(ProjectType.Product);
+            CreateProjectConfigReader($"TestResources\\SonarProjectConfig\\UnexpectedProjectTypeValue\\SonarProjectConfig.xml").ProjectType.Should().Be(ProjectType.Product);
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs:78]:
```suggestion
            CreateProjectConfigReader(@"TestResources\SonarProjectConfig\UnexpectedProjectTypeValue\SonarProjectConfig.xml").ProjectType.Should().Be(ProjectType.Product);
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs [hunk 12]
```diff
@@ -84,22 +84,22 @@ public void WhenHasMissingFilesToAnalyzePath_ReturnsCorrectValue()
         [DataRow("MissingProjectType")]
         [DataRow("MissingTargetFramework")]
         public void WhenHasMissingValues_FilesToAnalyzePath_ReturnsCorrectValue(string folder) =>
-            CreateProjectConfigReader($"ResourceTests\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml").FilesToAnalyzePath.Should().Be(@"c:\foo\bar\.sonarqube\conf\0\FilesToAnalyze.txt");
+            CreateProjectConfigReader($"TestResources\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml").FilesToAnalyzePath.Should().Be(@"c:\foo\bar\.sonarqube\conf\0\FilesToAnalyze.txt");
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs:87]:
```suggestion
            CreateProjectConfigReader($@"TestResources\SonarProjectConfig\{folder}\SonarProjectConfig.xml").FilesToAnalyzePath.Should().Be(@"c:\foo\bar\.sonarqube\conf\0\FilesToAnalyze.txt");
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs [hunk 13]
```diff
@@ -84,22 +84,22 @@ public void WhenHasMissingFilesToAnalyzePath_ReturnsCorrectValue()
...
 
         [DataTestMethod]
         [DataRow("Invalid_DifferentClassName")]
         [DataRow("Invalid_DifferentNamespace")]
         [DataRow("Invalid_Xml")]
         public void WhenInvalid_FilesToReturnPath_ThrowsException(string folder) =>
-            Assert.ThrowsException<InvalidOperationException>(() => CreateProjectConfigReader($"ResourceTests\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml"))
+            Assert.ThrowsException<InvalidOperationException>(() => CreateProjectConfigReader($"TestResources\\SonarProjectConfig\\{folder}\\SonarProjectConfig.xml"))
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTest.cs:94]:
```suggestion
            Assert.ThrowsException<InvalidOperationException>(() => CreateProjectConfigReader($@"TestResources\SonarProjectConfig\{folder}\SonarProjectConfig.xml"))
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarLintXmlTest.cs [hunk 14]
```diff
@@ -30,7 +30,7 @@ public class SonarLintXmlTest
     public void SonarLintXml_DeserializeFile_ExpectedValues()
     {
         var deserializer = new XmlSerializer(typeof(SonarLintXml));
-        using TextReader textReader = new StreamReader("ResourceTests\\SonarLintXml\\All_properties_cs\\SonarLint.xml");
+        using TextReader textReader = new StreamReader("TestResources\\SonarLintXml\\All_properties_cs\\SonarLint.xml");
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarLintXmlTest.cs:33]:
```suggestion
        using TextReader textReader = new StreamReader(@"TestResources\SonarLintXml\All_properties_cs\SonarLint.xml");
```

File: analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.cs (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextBaseTest.cs:181]: This one was needed

**Review by Tim-Pohlmann — APPROVED** [review]

**Pull request merged**